staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go

```diff
@@ -1607,21 +1521,8 @@ func TestNeedsUpdate(t *testing.T) {
 			testName: "If appProtocol is set to a different value",
 			updateFn: func() {
 				protocol := "http"
-				oldSvc = &v1.Service{
-					ObjectMeta: metav1.ObjectMeta{
-						Name:      "tcp-service",
-						Namespace: "default",
-					},
-					Spec: v1.ServiceSpec{
-						Ports: []v1.ServicePort{{
-							Port:        80,
-							Protocol:    v1.ProtocolTCP,
-							TargetPort:  intstr.Parse("22"),
-							AppProtocol: &protocol,
-						}},
-						Type: v1.ServiceTypeLoadBalancer,
-					},
-				}
+				oldSvc = newService("tcp-service", v1.ServiceTypeLoadBalancer, tweakAddPorts(v1.ProtocolTCP, 22))
+				oldSvc.Spec.Ports[0].AppProtocol = &protocol
```

Comment:
This should also be a tweak

Reply:
okay, I'll fix the unit test, and change here by the way